ecl/hql/hqlgram.y

```diff
@@ -8933,7 +8933,7 @@ simpleDataSet
                             IHqlExpression * arg = $7.getExpr();
                             if (isStringType(arg->queryType()))
                             {
-                                arg = createAttribute(nameAtom, arg);
+                                arg = createExprAttribute(nameAtom, arg);
```

Comment:
It spotted that code just below it in the same file was calling a createExprAttribute - and labelled it as a copy/paste error.  The changes only effect would be that non-constant names might work better.